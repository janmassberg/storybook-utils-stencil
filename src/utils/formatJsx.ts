import { format } from "prettier/standalone";
import parserBabel from "prettier/parser-babel";
import parserTypescript from "prettier/parser-typescript";

/**
 * Formats JSX source code.
 * @param {string} code The source code
 */
export const formatJsx = (code: string): string =>
    format(code, {
        parser: "typescript",
        plugins: [parserTypescript, parserBabel],
        htmlWhitespaceSensitivity: "ignore",
    });
