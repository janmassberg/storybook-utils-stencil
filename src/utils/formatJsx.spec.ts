import { formatJsx } from "./formatJsx";

describe("formatJsx", () => {
    it("returns correctly formatted JSX code", () => {
        expect(
            formatJsx(
                `<MyComponent
dataId="test-id"
dataName="test-name"
onClick={event => handleClick(event)}><span>test content</span></MyComponent>`
            )
        ).toMatchSnapshot();
    });
});
