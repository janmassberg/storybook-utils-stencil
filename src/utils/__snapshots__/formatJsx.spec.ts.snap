// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatJsx returns correctly formatted JSX code 1`] = `
"<MyComponent
  dataId=\\"test-id\\"
  dataName=\\"test-name\\"
  onClick={(event) => handleClick(event)}
>
  <span>test content</span>
</MyComponent>;
"
`;
